### PR TITLE
Notification setting models

### PR DESCRIPTION
**Pull request by KateMengjunXia**
https://nanopay.atlassian.net/browse/CPF-2940
nanopay side: https://github.com/nanoPayinc/NANOPAY/pull/8314

### CODE REVIEW

File: src/foam/nanos/notification/SendEmail.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        { name: 'notification', type: 'foam.nanos.notification.Notification' },
+      ],
+      javaCode: `
+      return;
+        
+      `
+    }
+  ]
+});
+
```
Comment (zahnno) [inline src/foam/nanos/notification/SendEmail.js]: Why is this model needed? Wouldn't the NotificationConfig implement the NotificationSetting?

Comment (jlhughes) [reply src/foam/nanos/notification/SendEmail.js]: This is one of the many NotificationSettings in the NotificationConfig.  The NotifcationConfig walks all settings and calls sentNotification()

Comment (jlhughes) [reply src/foam/nanos/notification/SendEmail.js]: @KateMengjunXia can this be named EmailNotificationSettings?
Also, where is the logic to walk through the settings and call sendNotification?

File: src/foam/nanos/notification/SendEmailSetting.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+foam.CLASS({
+  package: 'foam.nanos.notification',
+  name: 'SendEmailSetting',
```
Comment (jlhughes) [inline src/foam/nanos/notification/SendEmailSetting.js]: Why do we call these send...  why not just EmailSettings and NotificationSettings?
And the NotificationSettings has send(x, user, notification).

File: src/foam/nanos/auth/User.js [hunk 3]
```diff
@@ -589,6 +593,29 @@ foam.CLASS({
...
+      name: 'notify',
+      javaCode: `
+        DAO agentJunctionDAO      = (DAO) x.get("agentJunctionDAO");
+        DAO notificationConfigDAO = (DAO) x.get("notificationConfigDAO");
+        
+        UserUserJunction junction = (UserUserJunction) agentJunctionDAO
+          .find(EQ(UserUserJunction.SOURCE_ID, getId()));
+        if ( junction == null ) {
+          throw new RuntimeException("A junction between the user and its entities cannot be found.");
+        }
```
Comment (zahnno) [inline src/foam/nanos/auth/User.js]: User's should be calling their own notificationConfig, not one on the agentJunction - I think it'd be okay to have it exist on the user itself.

**Review by jlhughes — APPROVED** [review]
Looks good

**Pull request merged**